[12.0][MIG] account_invoice_force_number

tafaRU opened the pull request.

eLBati committed:
[ADD] account_invoice_force_number
moved from https://code.launchpad.net/~domsense/domsense-agilebg-addons/6.1

eLBati committed:
[FIX] header

[FIX] module author

[IMP] account_invoice_force_number does not work for supplier invoices

jgrandguillaume committed:
[MIGR] Set installable = False in modules to setup the 7.0 series

tafaRU committed:
[IMP] style and get installable

[FIX] authors' position

[REM] field name=type from views

[FIX] xpath values in invoice_view.xml

[FIX] PEP8 style

[FIX] PEP8 style: lines too long

Commit:
Launchpad automatic translations update.

pedrobaeza committed:
Fix PEP8

gurneyalex committed:
set addons as unported to 8.0

Commit:
Port to v8 with new API
Update POT file and add FR translation
More flexible xpath

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

Commit:
[FIX] duplicated field

sysadminmatmoz committed:
Slovenian translations added

eLBati committed:
[FIX] Attributes could be used directly instead of replace

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

oca-transbot committed:
OCA Transbot updated translations from Transifex

OpenCode committed:
Permission to set if an user can or can't use "Force Invoice Number"

Commit:
[IMP] test converted from yaml to unittest

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Rename manifest files

tafaRU committed:
[MIGR] account_invoice_force_number: preparatory tasks for migration and  apply OCA guidelines

tafaRU committed:
[MIGR] account_invoice_number: use move_name instead of internal_number

oca-transbot committed:
OCA Transbot updated translations from Transifex

OpenCode committed:
[MIG] account_invoice_force_number: Migration to 11.0

OpenCode committed:
[REM] Useless comment for utf-8 comments

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update account_invoice_force_number.pot

tafaRU committed:
[MIG] account_invoice_force_number: Migration to 12.0

tafaRU commented:
Please note I didn't squash the commits generated by Transifex as they have conflicts.

aitorbouzas commented on account_invoice_force_number/tests/__init__.py:
```diff
@@ -0,0 +1 @@
+from . import test_account_force_number
```
Missing new line.

tafaRU replied:
Done in https://github.com/OCA/account-invoicing/pull/434/commits/133ca19135a76f322a09376e8f893d35cdb034f0

aitorbouzas commented on account_invoice_force_number/tests/test_account_force_number.py:
```diff
@@ -0,0 +1,45 @@
+class TestAccountForceNumber(TransactionCase):
+
+    def setUp(self):
+        super(TestAccountForceNumber, self).setUp()
```
```super().method()``` should be used in Python 3.

tafaRU replied:
Done in https://github.com/OCA/account-invoicing/pull/434/commits/133ca19135a76f322a09376e8f893d35cdb034f0

aitorbouzas commented on account_invoice_force_number/tests/test_account_force_number.py:
```diff
@@ -0,0 +1,45 @@
+        invoice = self.env['account.invoice'].create({
+            'name': "Test Customer Invoice",
+            'journal_id': self.env['account.journal'].search(
+                [('type', '=', 'sale')])[0].id,
```
Isn't ```limit=1``` better in this case? You do it below.

tafaRU replied:
Done in https://github.com/OCA/account-invoicing/pull/434/commits/133ca19135a76f322a09376e8f893d35cdb034f0

aitorbouzas reviewed (changes requested):
Thanks for the migration! Code review.

tafaRU commented:
@aitorbouzas, could you please update your review?

astirpe commented on account_invoice_force_number/views/account_invoice_view.xml:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record model="ir.ui.view" id="invoice_supplier_form">
```
As per the [OCA Guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#format):

> Place id attribute before model

astirpe replied:
Could you also name this `id="..." `to (something like) `account_invoice_supplier_form_force_number` to make it consistent with the other one?

pedrobaeza replied:
This change is incorrect. Per guidelines, you should preserve the same ID (but different module), than original one

astirpe commented on account_invoice_force_number/views/account_invoice_view.xml:
```diff
@@ -0,0 +1,34 @@
+<odoo>
+
+    <record model="ir.ui.view" id="invoice_supplier_form">
+        <field name="name">account.invoice.supplier.form.force_number</field>
```
Since you are inheriting from another view, you could avoid to write the name here: it will be automatically generated.

astirpe commented on account_invoice_force_number/tests/test_account_force_number.py:
```diff
@@ -0,0 +1,45 @@
+class TestAccountForceNumber(TransactionCase):
+
+    def setUp(self):
+        super().setUp()
```
Do we need this? Looks like useless to me

astirpe commented on account_invoice_force_number/security/security.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
```
As per the [template](https://github.com/OCA/maintainer-tools/blob/master/template/module/security/some_model_security.xml#L1) you miss the encoding here

astirpe commented on account_invoice_force_number/tests/test_account_force_number.py:
```diff
@@ -0,0 +1,45 @@
+# Copyright 2016 Davide Corio - davidecorio.com
+# Copyright 2017 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp.tests.common import TransactionCase
```
`odoo`

astirpe commented on account_invoice_force_number/tests/test_account_force_number.py:
```diff
@@ -0,0 +1,45 @@
+        invoice.action_invoice_open()
+
+        # I check that the invoice number is the one we expect
+        self.assertEqual(invoice.number, invoice.move_name, msg='Wrong number')
```
Could you also add this line?

`self.assertEqual(invoice.number, '0001')`

 It implicitly checks that `invoice.move_name` is not modified when validating the invoice.

tafaRU commented:
@astirpe, thanks a lot for you review! I addressed your comments in ~https://github.com/OCA/account-invoicing/pull/434/commits/d31c13ec7e1a178e7a9a4120c87407d692d4b78a~ https://github.com/OCA/account-invoicing/pull/434/commits/204de9828f83ade5896001dc81d9aa020621acb1

astirpe reviewed (approved):
LGTM

emagdalenaC2i commented:
@pedrobaeza Are this ready to merge?

pedrobaeza commented:
@tafaRU maybe a bit of squashing by author/logical set?

eLBati commented:
@pedrobaeza rebased squashing and fixing up

primes2h commented:
@sbidoul 
The following is related to this merged module:
https://travis-ci.org/OCA/account-invoicing/builds/480270573

Transifex presence is a little bit suspicious :-)

sbidoul commented:
@primes2h the TRANSIFEX=1 step could be removed and replaced by MAKEPOT=1 on the odoo test line.

pedrobaeza commented:
But it's retro-compatible.

primes2h commented:
Here it is #481

Thank you!
